app/models/response.rb

```diff
@@ -32,30 +23,17 @@ class Response < ApplicationRecord
     end
   end
 
-  scope :recently_opened_and_not_sent, (lambda {
+  scope :recently_opened_and_not_invited, (lambda {
     where(
-      'open_from <= :time_now AND open_from > :recent_past AND invited_state = :not_sent',
+      'open_from <= :time_now AND open_from > :recent_past AND invitation_set_id IS NULL ' \
+      'AND completed_at IS NULL',
       time_now: Time.zone.now,
-      recent_past: RECENT_PAST.ago.in_time_zone,
-      not_sent: NOT_SENT_STATE
-    )
-  })
-
-  scope :still_open_and_not_completed, (lambda {
-    where(
-      'open_from <= :time_now AND open_from > :recent_past AND invited_state = :sent and completed_at IS NULL',
-      time_now: Time.zone.now - REMINDER_DELAY,
-      recent_past: (RECENT_PAST + REMINDER_DELAY).ago.in_time_zone,
-      sent: SENT_STATE
+      recent_past: RECENT_PAST.ago.in_time_zone
     )
   })
-
   scope :completed, (-> { where.not(completed_at: nil) })
   scope :invited, (lambda {
-    where('invited_state=? OR invited_state=? OR invited_state=?',
-          SENT_STATE,
-          SENDING_REMINDER_STATE,
-          REMINDER_SENT_STATE)
+    where('invitation_set_id IS NOT NULL')
```

Comment:
kan het niet zijn dat iemand nog niet geinvite is, maar al wel een invitation set heeft?

Reply:
Nou ja wat noem je geinvite. Dat alle invitations zijn verzonden? Dat de SMSjes zijn aangekomen op het mobiel? Het is moeilijk te checken. We maken een invitation_set aan die gooien we dan in een perform_later om in principe direct opgepakt te worden door de delayed job en dan de bijbehorende invitations uit te sturen. Anders moet je namelijk in de invitations zelf gaan kijken van welke invitation wel of niet verstuurd is en dat lijkt me wat te granulair/niet generiek.

Er kan idd een paar seconden tussen zitten dat we hier een response als invited zien en dat de gebruiker nog geen SMS of email heeft ontvangen maar dat lijkt me niet heel erg omdat je toch met onvoorspelbare delays zit in SMS en email.